tools/main.cpp

```diff
@@ -70,7 +70,18 @@ int main(int argc, char** argv) {
           antler::update_project,
           antler::validate_project> runner{app};
 
-   CLI11_PARSE(app, argc, argv);
+   try {
+      app.parse(argc, argv);
+   }
+   // This hack fix parsing error handling of an empty command argument. Example: antler-proj build ~/path/proj ""
+   catch (CLI::ExtrasError &e) {
+      if (std::string_view{"The following argument was not expected: "} == e.what()) {
+         return app.exit(CLI::ExtrasError("Empty argument is encountered in the command line", e.get_exit_code()));
+      }
+   }
```

Comment:
Missing an `app.exit(e)`.

Also, I agree with @mikelik - we should handle both "was" and "were". Here's a possibility.

Here's a suggestion:
 
```suggestion
   // Handle empty command argument. Example: `antler-proj build ~/path/proj ""`
   catch (CLI::ExtrasError& e) {
      std::string what = e.what();
      // trim space
      while(!what.empty() && what.back() == ' ')
         what.pop_back();
      if( what == "The following argument was not expected:" || what == "The following arguments were not expected:") {
         return app.exit(CLI::ExtrasError("Empty argument(s) encountered in the command line.", e.get_exit_code()));
      }
      app.exit(e);
   }
```

Reply:
Scott is right, if one would pass 2+ empty arguments (which is awkward but possible due to errors in scripts) - it will fail.
But I do not like his verbose solution. here is mine:

```suggestion
   // This hack fix parsing error handling of an empty command argument. Example: antler-proj build ~/path/proj ""
   catch (CLI::ExtrasError &e) {
      static const regex rex("The following argument(s)? (was|were)? not expected:\\s+$", std::regex_constants::ECMAScript);
    if (std::regex_search(e.what(), rex)) {
         return app.exit(CLI::ExtrasError("Unexpected empty argument is encountered in the command line", e.get_exit_code()));
      }
   }
```

Reply:
> Scott is right, if one would pass 2+ empty arguments (which is awkward but possible due to errors in scripts) - it will fail. But his solution seems to me too verbose and hard to understand from glance. here is mine:

I'm not against of such comprehensive check but if you try to test the cases with more than one empty argument you will see that they processed by the parser earlier and as I understand such exception  will be never  thrown for antler-proj command syntax. 
And I prefer Michal's and Scott's solution. Don't want to include complex libraries for such simple cases.